src/object/player.cpp

```diff
@@ -1863,10 +1863,10 @@ Player::set_bonus(BonusType type, bool animate)
     m_player_status.max_air_time[get_id()] = 0;
     m_player_status.max_earth_time[get_id()] = 0;
   }
-  if (type == FIRE_BONUS) m_player_status.max_fire_bullets[get_id()]++;
-  if (type == ICE_BONUS) m_player_status.max_ice_bullets[get_id()]++;
-  if (type == AIR_BONUS) m_player_status.max_air_time[get_id()]++;
-  if (type == EARTH_BONUS) m_player_status.max_earth_time[get_id()]++;
+  if (type == FIRE_BONUS && increment_powerup_counter) m_player_status.max_fire_bullets[get_id()]++;
+  if (type == ICE_BONUS && increment_powerup_counter) m_player_status.max_ice_bullets[get_id()]++;
+  if (type == AIR_BONUS && increment_powerup_counter) m_player_status.max_air_time[get_id()]++;
+  if (type == EARTH_BONUS && increment_powerup_counter) m_player_status.max_earth_time[get_id()]++;
```

Comment:
`increment_powerup_counter` condition could be moved to outer `if` containing other `if`s.

Reply:
Good point, this is indeed much more readable. Thanks.
I changed it. At least git tells me - github might be a bit laggy at the moment.